crates/db_views/src/custom_emoji_view.rs

```diff
@@ -35,16 +35,49 @@ impl CustomEmojiView {
     }
   }
 
-  pub async fn get_all(
+  pub async fn get_all(pool: &mut DbPool<'_>) -> Result<Vec<Self>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let emojis = custom_emoji::table
+      .left_join(
+        custom_emoji_keyword::table.on(custom_emoji_keyword::custom_emoji_id.eq(custom_emoji::id)),
+      )
+      .order(custom_emoji::category)
+      .then_order_by(custom_emoji::id)
+      .select((
+        custom_emoji::all_columns,
+        custom_emoji_keyword::all_columns.nullable(), // (or all the columns if you want)
+      ))
+      .load::<CustomEmojiTuple>(conn)
+      .await?;
+
+    Ok(CustomEmojiView::from_tuple_to_vec(emojis))
+  }
+
+  pub async fn list(
     pool: &mut DbPool<'_>,
-    for_local_site_id: LocalSiteId,
+    category: &Option<String>,
+    page: Option<i64>,
+    limit: Option<i64>,
+    ignore_page_limits: bool,
   ) -> Result<Vec<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
-    let emojis = custom_emoji::table
-      .filter(custom_emoji::local_site_id.eq(for_local_site_id))
+
+    let mut query = custom_emoji::table
       .left_join(
         custom_emoji_keyword::table.on(custom_emoji_keyword::custom_emoji_id.eq(custom_emoji::id)),
       )
+      .into_boxed();
+
+    if !ignore_page_limits {
+      let (limit, offset) = limit_and_offset(page, limit)?;
+      query = query.limit(limit).offset(offset);
+    }
+
+    if let Some(category) = category {
+      query = query.filter(custom_emoji::category.eq(category))
+    }
+
+    let emojis = query
       .order(custom_emoji::category)
```

Comment:
The category ordering should probably only happen if they supplied a category, just to be faster. So you could add `.order_by(category)` into the `if let Some(category)` above,  and add a `query = query.then_order_by(id)` below that block.